microprofile/WFLY-14281_Integrate-MP-Fault-Tolerance-3.0.adoc

```diff
@@ -0,0 +1,86 @@
+= Integrate MP Fault Tolerance 3.0 into EAP
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+MicroProfile 4.0 provides a major upgrade for https://github.com/eclipse/microprofile-fault-tolerance/releases/tag/3.0[MicroProfile Fault Tolerance 3.0] with changes described in its
+https://download.eclipse.org/microprofile/microprofile-fault-tolerance-3.0/microprofile-fault-tolerance-spec-3.0.html#release_notes_30[Release Notes].
+
+This component upgrade also covers the upgrade of SmallRye Fault Tolerance that implements MicroProfile Fault Tolerance 3.0.
+
+== Issue Metadata
+ 
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-14281[WFLY-14281 Integrate MP Fault Tolerance 3.0]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1594[EAP7-1594 Integrate MP Fault Tolerance 3.0 into EAP]
+
```

Comment:
As properly stated in the "Braking changes" section at the end of the proposal, shouldn't relevant Metrics related issues be mentioned here? 
BTW - Is the issue for Metrics update blocking this one? In such case mentioning this fact is also needed.
WDYT?

Reply:
Nothing in metrics is blocking us, it's just they needed to go in 'together'.

Reply:
From https://download.eclipse.org/microprofile/microprofile-fault-tolerance-3.0/microprofile-fault-tolerance-spec-3.0.html#release_notes_30:

```
The metrics added automatically by MicroProfile Fault Tolerance have been updated to take advantage of support for metric tags which was added to MicroProfile Metrics in version 2.0
```

So FT 3.0 uses Metrics 2.0 APIs, which I think is available already, see https://github.com/wildfly/wildfly/blob/master/pom.xml#L368.

Anyway, about blocking issues, you mean the only relatin is for the following specs:

https://projects.eclipse.org/projects/technology.microprofile/releases/4.0

to end up in WF 23 together, right?

Reply:
You need to look into the 'MP4' branch, so: https://github.com/wildfly/wildfly/blob/MP4/pom.xml#L369 which is already updated.

Yes, MP 4.0 needs to go in together as there are dependencies of specs on one another. There is no compatibility to take something from MP3 and use it with MP4 libs.

Reply:
Thanks. I looked at the `master` in order to double check that Metrics 2.* had been used already there.